File: frontend/src/types/universal_reports.ts
export interface StockValuationSummary {
    total_quantity: number;
    total_value: number;
    warehouse_count: number;
    item_count: number;
}

export interface ABCAnalysisItem {
    item_id: string;
    item_code: string;
    item_name: string;
    total_value_consumed: number;
    cumulative_percentage: number;
    classification: string;
}

export interface AgingReportItem {
    item_id: string;
    warehouse_id: string;
    quantity_on_hand: number;
    days_since_last_movement: number;
    aging_bucket: string;
}

File: backend/app/schemas/universal_reports.py
import uuid
from pydantic import BaseModel
from typing import List

class StockValuationSummary(BaseModel):
    total_quantity: float
    total_value: float
    warehouse_count: int
    item_count: int

class ABCAnalysisItem(BaseModel):
    item_id: uuid.UUID
    item_code: str | None = None
    item_name: str | None = None
    total_value_consumed: float
    cumulative_percentage: float
    classification: str # 'A', 'B', 'C'

class AgingReportItem(BaseModel):
    item_id: uuid.UUID
    warehouse_id: uuid.UUID
    quantity_on_hand: float
    days_since_last_movement: int
    aging_bucket: str # '0-30', '31-90', '91-180', '180+'

class KPIReport(BaseModel):
    total_inventory_value: float
    dead_stock_value: float
    active_batches: int
    items_near_expiry: int
